In [1]:
import pandas as pd
import datetime 
import numpy as np
import pickle
import matplotlib.pyplot as plt
import dtale
import re

In [2]:
smileon = pd.read_csv('../data/smileon_edited.csv')

In [3]:
pd.set_option('display.max_columns', 118)
smileon.head()

Unnamed: 0  patient_id  callreportnum reportversion        callstart  \
0           0     3638169       93115324  registration  9/30/2021 16:56   
1           1     4431194       93115188  registration  9/30/2021 16:53   
2           2     2941341       93115081      clinical  9/30/2021 16:50   
3           3     3823933       93115032      clinical  9/30/2021 16:48   
4           4     3348256       93114968      clinical  9/30/2021 16:46   

           callend  calllength  callstaff_id  entrystaff_id        entrydate  \
0  9/30/2021 16:57           1        123747         123747  9/30/2021 16:57   
1  9/30/2021 16:53           0        123747         123747  9/30/2021 16:53   
2  9/30/2021 16:51           1        161979         161979  9/30/2021 16:51   
3  9/30/2021 16:49           1        161979         161979  9/30/2021 16:49   
4  9/30/2021 16:48           2        161979         161979  9/30/2021 16:48   

  Reviewed ReferralsMade patient_bday       city      county state      zip  \
0       No           NaN         1960     Smyrna  Rutherford    TN  37167.0   
1       No           NaN         1959  Nashville    Davidson    TN  37211.0   
2       No           NaN         1942    Memphis      Shelby    TN  38104.0   
3       No           NaN         1960    Memphis      Shelby    TN  38109.0   
4       No           NaN         1955    Memphis      Shelby    TN      NaN   

  demographic_household demographic_lang demographic_preferredlang  \
0                   1.0              NaN                   English   
1                   1.0              NaN                   English   
2                   1.0              NaN                       NaN   
3                   1.0              NaN                       NaN   
4                   1.0              NaN                       NaN   

  demographic_ethnic demographic_gender demographic_income  \
0   African American             Female       $0 - $21 280   
1          Caucasian             Female       $0 - $21 280   
2              Asian               Male       $0 - $21 280   
3   African American               Male       $0 - $21 280   
4   African American             Female       $0 - $21 280   

  demographic_howyouhearprogram demographic_otherassistance  \
0                         Other              SNAP; SSI/SSDI   
1           Dental Clinic Staff                         NaN   
2           Dental Clinic Staff                         NaN   
3           Dental Clinic Staff                         NaN   
4           Dental Clinic Staff                    SSI/SSDI   

  demographic_lastdentistvisit pe_insurance pe_smileonprequalify  \
0                          NaN           No                  Yes   
1            More than 5 years           No                  Yes   
2          Less than 12 months           No                  Yes   
3            More than 5 years           No                  Yes   
4                 3 to 5 years           No                  Yes   

                 pe_nextstep               pe_nextstep2 ve_transitarranged  \
0  Continued with Navigation                        NaN                NaN   
1  Continued with Navigation                        NaN                NaN   
2                        NaN  Continued with Navigation                NaN   
3                        NaN  Continued with Navigation                NaN   
4                        NaN  Continued with Navigation                NaN   

  ve_eligibility ve_verification_date  \
0            NaN                  NaN   
1            NaN                  NaN   
2            NaN            10/8/2019   
3            NaN           11/10/2020   
4            NaN            4/15/2020   

                                    si_clinicname si_typewhereclientenrolled  \
0                                             NaN                        NaN   
1                                             NaN         Referral Follow-Up   
2  Christ Community Health Services- Broad Avenue         

In [4]:
smileon = smileon.sort_values(['patient_id', 'entrydate'])

In [5]:
smileon = smileon.reset_index()
smileon.head(70)

index  Unnamed: 0  patient_id  callreportnum reportversion  \
0   13186       13186          -2       84708447  registration   
1   47543       47543          -2       62945363      clinical   
2   42846       42846          -2       65438351  registration   
3     228         228          -2       92968098      clinical   
4     229         229          -2       92967776  registration   
..    ...         ...         ...            ...           ...   
65  39680       39680          -1       66703310      clinical   
66  39677       39677          -1       66703536      clinical   
67  39675       39675          -1       66703613      clinical   
68  39673       39673          -1       66703692      clinical   
69  32182       32182     2384106       70281146      clinical   

          callstart          callend  calllength  callstaff_id  entrystaff_id  \
0   2/19/2021 12:10  2/19/2021 12:56          46        119513         119513   
1   5/15/2019 10:00  5/15/2019 11:00          60        119473         119473   
2    8/13/2019 8:00   8/13/2019 8:05           5        133281         133281   
3   9/27/2021 12:56  9/27/2021 12:56           0        140162         140162   
4   9/27/2021 12:51  9/27/2021 12:56           5        140162         140162   
..              ...              ...         ...           ...            ...   
65   9/27/2019 9:17   9/27/2019 9:18           1        118843         118843   
66   9/27/2019 9:26   9/27/2019 9:27           1        118843         118843   
67   9/27/2019 9:28   9/27/2019 9:29           1        118843         118843   
68   9/27/2019 9:31   9/27/2019 9:32           1        118843         118843   
69  1/20/2020 13:25  1/20/2020 13:25           0        134549         134549   

          entrydate Reviewed ReferralsMade patient_bday            city  \
0   2/19/2021 12:56       No           NaN         1954    Johnson City   
1   5/15/2019 10:05       No           NaN         1953             NaN   
2    8/13/2019 8:05       No           NaN         1957         Memphis   
3   9/27/2021 12:56       No           NaN         1957  Mount Pleasant   
4   9/27/2021 12:56       No           NaN         1957  Mount Pleasant   
..              ...      ...           ...          ...             ...   
65   9/27/2019 9:18       No           NaN         1900             NaN   
66   9/27/2019 9:27       No           NaN         1900             NaN   
67   9/27/2019 9:30       No           NaN         1900             NaN   
68   9/27/2019 9:32       No           NaN         1900             NaN   
69  1/27/2020 13:25       No           NaN         1944    Murfreesboro   

        county state      zip demographic_household demographic_lang  \
0   Washington    TN  37604.0                   4.0              NaN   
1          NaN   NaN      NaN                     2              NaN   
2       Shelby    TN  38127.0                     1              NaN   
3        Maury    TN      NaN                   4.0              NaN   
4        Maury    TN      NaN                   4.0              NaN   
..         ...   ...      ...                   ...              ...   
65  Rutherford    TN      NaN                   NaN              NaN   
66  Rutherford    TN      NaN                   NaN              NaN   
67  Rutherford    TN      NaN                   NaN              NaN   
68  Rutherford    TN      NaN                   NaN              NaN   
69  Rutherford    TN  37130.0                     5              NaN   

   demographic_preferredlang demographic_ethnic demographic_gender  \
0                    Spanish           Hispanic             Female   
1                        NaN          Caucasian               Male   
2                    English   African American             Female   
3                        NaN          Caucasian             Female   
4                    English          Caucasian             Female   
..                       ...                ...     

In [6]:
#dropping the first 69 rows (rows 0-68) because they're weird - looks like test entries or something?  
#all data for patient's # -1 and -2
smileon = smileon.drop(range(0,69))
smileon = smileon.reset_index(drop=True)
smileon.head(10)

index  Unnamed: 0  patient_id  callreportnum reportversion  \
0  32182       32182     2384106       70281146      clinical   
1  53004       53004     2384106       59821075      clinical   
2  34169       34169     2384106       69642939      clinical   
3  56013       56013     2384106       57055663      clinical   
4  38027       38027     2384106       67492062      clinical   
5  55239       55239     2384106       57578517      clinical   
6  35960       35960     2384106       68297290      clinical   
7  53517       53517     2384106       59037548      clinical   
8  54518       54518     2384106       58526759      clinical   
9  49813       49813     2384106       61675197      clinical   

          callstart           callend  calllength  callstaff_id  \
0   1/20/2020 13:25   1/20/2020 13:25           0        134549   
1    1/9/2019 10:07    1/9/2019 10:07           0        118821   
2  12/16/2019 10:35  12/16/2019 10:35           0        134549   
3   10/9/2018 18:59   10/9/2018 18:59           0        118821   
4   10/21/2019 8:35   10/21/2019 8:35           0        134549   
5  10/30/2018 17:09  10/30/2018 17:10           1        118821   
6  11/18/2019 10:26  11/18/2019 10:27           1        134549   
7  12/17/2018 15:12  12/17/2018 15:12           0        118821   
8  11/19/2018 12:35  11/19/2018 12:36           1        118821   
9   3/26/2019 10:55   3/26/2019 10:56           1        116749   

   entrystaff_id         entrydate Reviewed ReferralsMade patient_bday  \
0         134549   1/27/2020 13:25       No           NaN         1944   
1         118821   1/28/2019 10:07       No           NaN         1944   
2         134549    1/6/2020 10:35       No           NaN         1944   
3         118821  10/11/2018 18:59       No           NaN         1944   
4         134549   10/24/2019 8:35       No           NaN         1944   
5         118821  10/31/2018 17:10       No           NaN         1944   
6         134549  11/19/2019 10:27       No           NaN         1944   
7         118821  12/27/2018 15:12       No           NaN         1944   
8         118821   12/6/2018 12:36       No           NaN         1944   
9         116749   3/29/2019 10:56       No           NaN         1944   

           city      county state      zip demographic_household  \
0  Murfreesboro  Rutherford    TN  37130.0                     5   
1  Murfreesboro  Rutherford    TN  37130.0                     5   
2  Murfreesboro  Rutherford    TN  37130.0                     5   
3  Murfreesboro  Rutherford    TN  37130.0                     5   
4  Murfreesboro  Rutherford    TN  37130.0                     5   
5  Murfreesboro  Rutherford    TN  37130.0                     5   
6  Murfreesboro  Rutherford    TN  37130.0                     5   
7  Murfreesboro  Rutherford    TN  37130.0                     5   
8  Murfreesboro  Rutherford    TN  37130.0                     5   
9  Murfreesboro  Rutherford    TN  37130.0                     5   

  demographic_lang demographic_preferredlang demographic_ethnic  \
0              NaN                    Arabic            Unknown   
1              NaN                    Arabic            Unknown   
2              NaN                    Arabic            Unknown   
3              NaN                    Arabic            Unknown   
4              NaN                    Arabic            Unknown   
5              NaN                    Arabic            Unknown   
6              NaN                    Arabic            Unknown   
7              NaN                    Arabic            Unknown   
8              NaN                    Arabic            Unknown   
9              NaN                    Arabic            Unknown   

  demographic_gender demographic_income demographic_howyouhearprogram  \
0             Female       $0 - $21 280           Dental Clinic Staff   
1             Female       $0 - $21 280           Dental Clinic Staff   
2             Female       $

In [7]:
smileon['patient_id'].value_counts()

3212379    54
2441776    43
2758373    42
2431132    37
2416076    35
           ..
2628473     1
2530943     1
2511760     1
2635370     1
2823981     1
Name: patient_id, Length: 10709, dtype: int64

In [8]:
#NOTE :  THERE ARE 10709 UNIQUE PATIENTS IN THIS DF

In [9]:
smileon['reportversion'].value_counts()

clinical        40173
registration    17618
Name: reportversion, dtype: int64

In [10]:
#NOTE: BUT THERE ARE 17622 REGISTRATIONS.  MEANINS SOME PATIENTS, THE SAME PATIENT EVEN, REGISTERED MORE THAN ONCE

In [11]:
#I would really like to be able to see all the columns.  Going back to fix that
smileon.loc[smileon['patient_id']==2416076]

index  Unnamed: 0  patient_id  callreportnum reportversion  \
2477  32756       32756     2416076       70080607      clinical   
2478  33854       33854     2416076       69737230      clinical   
2479  55777       55777     2416076       57225628      clinical   
2480  56291       56291     2416076       56828901      clinical   
2481  56129       56129     2416076       56850717      clinical   
2482  18823       18823     2416076       80792354      clinical   
2483  36240       36240     2416076       68190928      clinical   
2484  54987       54987     2416076       57786508      clinical   
2485  54774       54774     2416076       58628791      clinical   
2486  54503       54503     2416076       58629436      clinical   
2487  54357       54357     2416076       58537949      clinical   
2488  54167       54167     2416076       58538015      clinical   
2489  17162       17162     2416076       81805334      clinical   
2490  30948       30948     2416076       70810955      clinical   
2491  51798       51798     2416076       60512593      clinical   
2492  50795       50795     2416076       61093830      clinical   
2493  50352       50352     2416076       61556184      clinical   
2494  11642       11642     2416076       86265555      clinical   
2495  49694       49694     2416076       61756634      clinical   
2496   9589        9589     2416076       87404088      clinical   
2497  56257       56257     2416076       62871704      clinical   
2498  47452       47452     2416076       62975344      clinical   
2499   7849        7849     2416076       88630866      clinical   
2500  47051       47051     2416076       63825431      clinical   
2501  46343       46343     2416076       63825491      clinical   
2502   3947        3947     2416076       90710157      clinical   
2503  45782       45782     2416076       64344056      clinical   
2504   5050        5050     2416076       89956332      clinical   
2505  42007       42007     2416076       65850976      clinical   
2506  57159       57159     2416076       55875397  registration   
2507  57157       57157     2416076       55875620      clinical   
2508  44024       44024     2416076       65220398      clinical   
2509  23287       23287     2416076       77329572      clinical   
2510  21789       21789     2416076       78500974      clinical   
2511  40806       40806     2416076       66380005      clinical   

             callstart           callend  calllength  callstaff_id  \
2477   1/13/2020 12:49   1/13/2020 12:52           3        117268   
2478  12/19/2019 11:37  12/19/2019 11:38           1        116749   
2479  10/17/2018 10:15  10/17/2018 10:16           1        118821   
2480   10/2/2018 13:47   10/2/2018 13:49           2        118821   
2481    10/5/2018 9:46    10/5/2018 9:55           9        118820   
2482  10/26/2020 14:05  10/26/2020 14:08           3        134549   
2483  11/13/2019 10:17  11/13/2019 10:18           1        134549   
2484    11/7/2018 9:26    11/7/2018 9:27           1        118821   
2485  11/12/2018 15:32  11/12/2018 15:33           1        118821   
2486  11/19/2018 15:50  11/19/2018 15:50           0        118821   
2487  11/26/2018 18:04  11/26/2018 18:05           1        118821   
2488  11/29/2018 18:07  11/29/2018 18:07           0        118821   
2489  11/25/2020 13:45  11/25/2020 13:46           1        134549   
2490     2/5/2020 9:54     2/5/2020 9:54           0        134549   
2491   2/13/2019 10:53   2/13/2019 10:53           0        118821   
2492    3/6/2019 12:20    3/6/2019 12:21           1        118821   
2493   3/14/2019 14:08   3/14/2019 14:08           0        118821   
2494   3/18/2021 12:58   3/18/2021 13:02           4        134549   
2495   3/28/2019 11:43   3/28/2019 11:43           0        116749   
2496   4/22/2021 10:19   4/22/2021 10:20           1        134549   
2497   10/2/2018 17:46   10/2/2018 17:46           0        121620   
2498   5/16/2019 

In [12]:
#my df did not organize quite as well as i'd hoped - try again?
smileon = smileon.sort_values(['patient_id', 'callreportnum'])

#So, i was hoping that i'd see a row of needed procedures and then see them drop off over time to show progress ... but that's not what i have here.  What i have is procedures that were done.  And, i guess, how many times they were done.  Like oce_2330 - that's "resin one surface, anterior".  and sometimes it's in this patient's row x1, others x6.  so - i'm guessing that means they had six teeth done, with resin applied to one surface per tooth.

so, i can see that, with each visit, the patient had more work done.  and in the end, their plan was not completed and their full function was not restored.  Does that mean they weren't served ... ?
Obviously - no.  they were in better condition when they left than when they arrived - we see a ton of work done

Number of procedures per patient is a poor metric of success.  some need more help than others.  

perhaps a window could be - how many of these procedures are being offered overall.  a sum of each column.  finding the maximum column.  totalling some of them together since they're basically the same thing - veneers for example, irresespective of how many sides of the tooth.  that's important for billing - but not for treatment stats.

another window could be what complaint a person has when they arrive, and what treatments they received.  That will be harder to determine - but could be interesting.  Let's take a look at the complaints column.

In [13]:
smileon.head()

index  Unnamed: 0  patient_id  callreportnum reportversion  \
11  57818       57818     2384106       55083365  registration   
13  57743       57743     2384106       55264053      clinical   
12  57851       57851     2384106       55628771      clinical   
3   56013       56013     2384106       57055663      clinical   
5   55239       55239     2384106       57578517      clinical   

           callstart           callend  calllength  callstaff_id  \
11    8/1/2018 13:22    8/1/2018 14:09          47        118833   
13    8/7/2018 10:53    8/7/2018 10:55           2        116749   
12     8/1/2018 8:53     8/1/2018 8:54           1        118821   
3    10/9/2018 18:59   10/9/2018 18:59           0        118821   
5   10/30/2018 17:09  10/30/2018 17:10           1        118821   

    entrystaff_id         entrydate Reviewed ReferralsMade patient_bday  \
11         118833   7/31/2018 14:09       No           NaN         1944   
13         116749    8/7/2018 10:55       No           NaN         1944   
12         118821    8/21/2018 8:54       No           NaN         1944   
3          118821  10/11/2018 18:59       No           NaN         1944   
5          118821  10/31/2018 17:10       No           NaN         1944   

            city      county state      zip demographic_household  \
11  Murfreesboro  Rutherford    TN  37130.0                     5   
13  Murfreesboro  Rutherford    TN  37130.0                     5   
12  Murfreesboro  Rutherford    TN  37130.0                     5   
3   Murfreesboro  Rutherford    TN  37130.0                     5   
5   Murfreesboro  Rutherford    TN  37130.0                     5   

   demographic_lang demographic_preferredlang demographic_ethnic  \
11              NaN                    Arabic                NaN   
13              NaN                    Arabic            Unknown   
12              NaN                    Arabic            Unknown   
3               NaN                    Arabic            Unknown   
5               NaN                    Arabic            Unknown   

   demographic_gender demographic_income demographic_howyouhearprogram  \
11             Female       $0 - $21 280           Dental Clinic Staff   
13             Female       $0 - $21 280           Dental Clinic Staff   
12             Female       $0 - $21 280           Dental Clinic Staff   
3              Female       $0 - $21 280           Dental Clinic Staff   
5              Female       $0 - $21 280           Dental Clinic Staff   

   demographic_otherassistance demographic_lastdentistvisit pe_insurance  \
11              SNAP; TennCare                 3 to 5 years           No   
13              SNAP; TennCare                 3 to 5 years           No   
12              SNAP; TennCare                 3 to 5 years           No   
3               SNAP; TennCare                 3 to 5 years           No   
5               SNAP; TennCare                 3 to 5 years           No   

   pe_smileonprequalify                pe_nextstep               pe_nextstep2  \
11                  Yes  Continued with Navigation                        NaN   
13                  Yes                        NaN  Continued with Navigation   
12                  Yes                        NaN  Continued with Navigation   
3                   Yes                        NaN  Continued with Navigation   
5                   Yes                        NaN  Continued with Navigation   

   ve_transitarranged ve_eligibility ve_verification_date  \
11                NaN            NaN                  NaN   
13                NaN            Yes             8/1/2018   
12                NaN            NaN             8/1/2018   
3                 NaN            NaN             8/1/2018   
5                 NaN            NaN             8/1/2018   

                              si_clinicname si_typewhereclientenrolled  \
11  Interfaith Dental Clinic - Murfreesboro                        NaN   
13                 

In [14]:
smileon['s_problems12months'].value_counts()

Difficulty when biting or chewing foods                                                                                                                                                468
Experienced pain or discomfort                                                                                                                                                         381
Difficulty when biting or chewing foods; Experienced pain or discomfort                                                                                                                362
Dry mouth                                                                                                                                                                              344
Difficulty when biting or chewing foods; Difficulty with speech or pronouncing words; Dry mouth; Felt anxiety; Felt embarrassement; Avoided smiling; Experienced pain or discomfort    317
                                                                 

#okay - they warned about that.  there's only 7 options for entry, but they're not broken up here.  There's a lot of possible combos.  

so, i need to break this up into 7 columns, and populate each with a 1 if that phrase exists in the 12months complaints column.  probably a regex function. they do at least show in order ... so, if a person is having problems biting foods, that will always be first in the list.

In [15]:
#let's see about some stats:
smileon.describe()

index    Unnamed: 0    patient_id  callreportnum    calllength  \
count  57791.000000  57791.000000  5.779100e+04   5.779100e+04  57791.000000   
mean   28920.357997  28920.357997  3.083800e+06   7.417992e+07      4.097541   
std    16707.115214  16707.115214  5.808006e+05   1.086729e+07     10.497058   
min        0.000000      0.000000  2.384106e+06   5.508336e+07      0.000000   
25%    14452.500000  14452.500000  2.633565e+06   6.540958e+07      1.000000   
50%    28901.000000  28901.000000  2.891697e+06   7.157117e+07      1.000000   
75%    43409.500000  43409.500000  3.498203e+06   8.423406e+07      3.000000   
max    57859.000000  57859.000000  4.581581e+06   9.350847e+07    735.000000   

        callstaff_id  entrystaff_id           zip  s_num_lowerteeth  \
count   57791.000000   57791.000000  54787.000000       6183.000000   
mean   124761.271201  124851.273036  37551.853852         10.084910   
std     14704.883192   14248.340910    507.528006          3.183853   
min        -1.000000   60132.000000  24201.000000          1.000000   
25%    118845.000000  118845.000000  37146.000000          8.000000   
50%    123747.000000  123747.000000  37334.000000         10.000000   
75%    134546.000000  134546.000000  38018.000000         12.000000   
max    164262.000000  164262.000000  38654.000000         16.000000   

       s_num_upperteeth      oce_230   oce_1351    oce_1354   oce_2140  \
count       5299.000000  1337.000000  12.000000  891.000000  50.000000   
mean           9.824495     2.549738   2.416667    3.244669   1.280000   
std            3.537183     2.173077   2.353270    3.039194   1.050559   
min            1.000000     1.000000   1.000000    1.000000   1.000000   
25%            7.000000     1.000000   1.000000    1.000000   1.000000   
50%           10.000000     2.000000   1.000000    2.000000   1.000000   
75%           13.000000     3.000000   2.500000    4.000000   1.000000   
max           16.000000    17.000000   8.000000   22.000000   8.000000   

         oce_2150   oce_2160  oce_2161    oce_2330    oce_2331    oce_2332  \
count  104.000000  46.000000      10.0  944.000000  940.000000  579.000000   
mean     1.057692   1.043478       1.0    1.507415    1.361702    1.255613   
std      0.234290   0.206185       0.0    1.027691    1.062990    0.628297   
min      1.000000   1.000000       1.0    1.000000    1.000000    1.000000   
25%      1.000000   1.000000       1.0    1.000000    1.000000    1.000000   
50%      1.000000   1.000000       1.0    1.000000    1.000000    1.000000   
75%      1.000000   1.000000       1.0    2.000000    2.000000    1.000000   
max      2.000000   2.000000       1.0   10.000000   27.000000    8.000000   

         oce_2335  oce_2390     oce_2391     oce_2392    oce_2394  oce_2915  \
count  428.000000       8.0  1226.000000  1341.000000  171.000000       2.0   
mean     1.343458       1.0     1.273246     1.158091    1.116959       1.0   
std      0.769001       0.0     0.579558     0.524394    0.403374       0.0   
min      1.000000       1.0     1.000000     1.000000    1.000000       1.0   
25%      1.000000       1.0     1.000000     1.000000    1.000000       1.0   
50%      1.000000       1.0     1.000000     1.000000    1.000000       1.0   
75%      1.000000       1.0     1.000000     1.000000    1.000000       1.0   
max      9.000000       1.0     5.000000    14.000000    4.000000       1.0   

         oce_2920   oce_2933   oce_2940  oce_2954  oce_2980  oce_3220  \
count  134.000000  19.000000  74.000000      15.0       3.0       7.0   
mean     1.037313   1.052632   1.108108       1.0       1.0       1.0   
std      0.190240   0.229416   0.424186       0.0       0.0       0.0   
min      1.000000   1.000000   1.000000       1.0       1.0       1.0   
25%      1.000000   1.000000   1.000000       1.0       1.0       1.0   
50%      1.000000   1.000000   1.000000       1.0       1.0       1.0   
75%      1.000000   1.000000   1.000000       1.0       1.0 

oce_230:   x-rays          1337
oce_1351:  tooth sealant   12
oce_1354:  topical med     891
oce_2140:  filling 1s      50
oce_2150:  filling 2s      104
oce_2160:  filling 3s      46
oce_2161:  filling 4s      10
        TOTAL FILLINGS:        210
oce_2330:  resin 1s        944
oce_2331:  resin 2s        940
oce_2332:  resin 3s        579
oce_2335:  resin 4s        428
oce_2390:  crown           8
oce_2391:  postresin 1s    1226
oce_2392:  postresin 2s    1341
oce_2394:  postresin 3s    171
oce_2915:  prosthetic cast 2
oce_2920:  pros. crown     134
oce_2933:  steel crown     19
oce_2940:  sed. filling    74
oce_2954:  toothrebuild    15
oce_2980:  crown repair    3
oce_3220:  save rot.tooth  7
oce_4341:  plaque removal  1396
oce_6930:  recement bridge 12
oce_6980:  repair bridge   2
oce_7111:  extraction rem. 9
oce_7210:  surg. extract   819
oce_7220:  removal impacts 5
oce_7230:  removal impactpb 6
oce_7240:  removal impactcb 19
oce_7241:  removal complica 3
oce_7250:  surgery roots   486
oce_7261:  sinus closure   1
oce_7286:  biopsy          68
oce_7310:  alveoloplasty   225
oce_7311:  alveoloplasty   30
oce_7321:  alveoloplasty   80
oce_7410:  excision b.lesion 16
oce_7411:  excision b.lesion 1
oce_7471:  removal exostosis 4
oce_7510:  abscess drain   12
oce_7971:  pericoronal gin.  2
oce_7972:  surg. fibroustub. 1
oce_9110:  emergency treatm. 26
oce_9951:  occlusal adjust.  83

TOTAL cosmetic RESINS/RESTORATIONS:  5637
TOTAL SURGERIES:            1696

most common treatment: resin     2738
#2 most common:  plaque removal  1396
#3 most common:  xrays           1337

In [16]:
smileon.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 57791 entries, 11 to 57790
Columns: 118 entries, index to tp_clientrestored
dtypes: float64(48), int64(7), object(63)
memory usage: 52.5+ MB


In [17]:
#okay - let's figure out how to break apart that complaints column into 7 different columns, one for each:
#s_problems12months
s_difficulty_eating = []
s_difficulty_speaking = []
s_drymouth = []
s_felt_anxiety = []
s_felt_embarrassed = []
s_avoided_smiling = []
s_pain_discomfort = []

In [18]:
#smileon[smileon['s_problems12months'].str.contains(r'(\w*\s+)chewing(\s+\w*)')]

In [19]:
#looks like i can't filter this column yet because of the nan values.  so - can i drop them, then maybe merge things back in?
smileon_drop = smileon.dropna(subset=['s_problems12months'])

In [23]:
smileon_drop['s_difficulty_eating'] = 0
smileon_drop['s_difficulty_speaking'] = 0
smileon_drop['s_drymouth'] = 0
smileon_drop['s_felt_anxiety'] = 0
smileon_drop['s_felt_embarrassed'] = 0
smileon_drop['s_avoided_smiling'] = 0
smileon_drop['s_pain_discomfort'] = 0

In [24]:
smileon_drop.head()

index  Unnamed: 0  patient_id  callreportnum reportversion  \
13  57743       57743     2384106       55264053      clinical   
19  57842       57842     2384744       55107015      clinical   
34  57667       57667     2384833       55320630      clinical   
39  57744       57744     2384922       55263902      clinical   
59  57335       57335     2384932       55699959      clinical   

          callstart          callend  calllength  callstaff_id  entrystaff_id  \
13   8/7/2018 10:53   8/7/2018 10:55           2        116749         116749   
19   8/1/2018 10:00   8/1/2018 10:22          22        118820         118820   
34    8/9/2018 8:49    8/9/2018 8:50           1        118820         118820   
39   8/7/2018 10:49   8/7/2018 10:50           1        116749         116749   
59  8/23/2018 14:37  8/23/2018 15:05          28        118833         118833   

          entrydate Reviewed ReferralsMade patient_bday          city  \
13   8/7/2018 10:55       No           NaN         1944  Murfreesboro   
19   8/1/2018 10:22       No           NaN         1954     Nashville   
34    8/9/2018 8:50       No           NaN         1942    Hartsville   
39   8/7/2018 10:50       No           NaN         1937  Murfreesboro   
59  8/23/2018 15:05       No           NaN         1952  Murfreesboro   

        county state      zip demographic_household demographic_lang  \
13  Rutherford    TN  37130.0                     5              NaN   
19    Davidson    TN  37211.0                     1              NaN   
34   Trousdale    TN  37074.0                     1              NaN   
39  Rutherford    TN  37129.0                     1              NaN   
59  Rutherford    TN  37129.0                     2              NaN   

   demographic_preferredlang demographic_ethnic demographic_gender  \
13                    Arabic            Unknown             Female   
19                       NaN          Caucasian             Female   
34                       NaN          Caucasian               Male   
39                       NaN                NaN               Male   
59                   Spanish           Hispanic               Male   

   demographic_income demographic_howyouhearprogram  \
13       $0 - $21 280           Dental Clinic Staff   
19      $21246-$24280                This encounter   
34                NaN           Dental Clinic Staff   
39       $0 - $21 280           Dental Clinic Staff   
59       $0 - $21 280           Dental Clinic Staff   

   demographic_otherassistance demographic_lastdentistvisit pe_insurance  \
13              SNAP; TennCare                 3 to 5 years           No   
19                         NaN          Less than 12 months           No   
34                         NaN          Less than 12 months           No   
39                         NaN          Less than 12 months           No   
59                         NaN                 3 to 5 years           No   

   pe_smileonprequalify pe_nextstep               pe_nextstep2  \
13                  Yes         NaN  Continued with Navigation   
19                  Yes         NaN  Continued with Navigation   
34                  Yes         NaN  Continued with Navigation   
39                  Yes         NaN  Continued with Navigation   
59                  Yes         NaN  Continued with Navigation   

   ve_transitarranged ve_eligibility ve_verification_date  \
13                NaN            Yes             8/1/2018   
19                 No            Yes             8/1/2018   
34                NaN            NaN             8/7/2018   
39                 No            Yes             8/1/2018   
59                NaN            Yes            8/23/2018   

                              si_clinicname si_typewhereclientenrolled  \
13                                      NaN                        NaN   
19                                      NaN                        NaN   
34                                      NaN          

In [20]:
regex1 = smileon_drop[smileon_drop['s_problems12months'].str.contains(r'(\w*\s+)chewing(\s+\w*)')]  #3385
regex2 = smileon_drop[smileon_drop['s_problems12months'].str.contains(r'(\w*\s+)speech(\s+\w*)')]  #1348
regex3 = smileon_drop[smileon_drop['s_problems12months'].str.contains(r'(\w*\s+)Dry(\s+\w*)')]  #1454
regex4 = smileon_drop[smileon_drop['s_problems12months'].str.contains(r'(\w*\s+)anxiety(\S+\w*)')]  #1447
regex5 = smileon_drop[smileon_drop['s_problems12months'].str.contains(r'(\w*\s+)embarrassement(\S+\w*)')]  #1894
regex6 = smileon_drop[smileon_drop['s_problems12months'].str.contains(r'(\w*\s+)smiling(\S+\w*)')]  #1394
regex7 = smileon_drop[smileon_drop['s_problems12months'].str.contains(r'(\w*\s+)pain(\s+\w*)')]  #2835

In [21]:
#okay! that's working - now how to get these regex filters to populate the empty libaries with "1" or "0" values ... 
#and how to get those columns in the df?

In [28]:
'''for x in smileon_drop['s_problems12months']:
    if regex1:
        smileon_drop['s_difficulty_eating'] = 1'''
#of course that's too easy ... try again

'''for x in smileon_drop['s_problems12months']:
    if smileon_drop['s_problems12months'].str.contains(r'(\w*\s+)chewing(\s+\w*)'):
        smileon_drop['s_difficulty_eating'] = 1'''
#still nope

ValueError: The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().

In [39]:
smileon_drop['s_difficulty_eating'] = pd.np.where(smileon_drop['s_problems12months'].str.contains("chewing"),1,0)
#this code is working well!  looks like i don't even need regex at all ... 
smileon_drop['s_difficulty_speaking'] = pd.np.where(smileon_drop['s_problems12months'].str.contains("speech"),1,0)
smileon_drop['s_drymouth'] = pd.np.where(smileon_drop['s_problems12months'].str.contains("Dry"),1,0)
smileon_drop['s_felt_anxiety'] = pd.np.where(smileon_drop['s_problems12months'].str.contains("anxiety"),1,0)
smileon_drop['s_felt_embarrassed'] = pd.np.where(smileon_drop['s_problems12months'].str.contains("embarrassement"),1,0)
smileon_drop['s_avoided_smiling'] = pd.np.where(smileon_drop['s_problems12months'].str.contains("smiling"),1,0)
smileon_drop['s_pain_discomfort'] = pd.np.where(smileon_drop['s_problems12months'].str.contains("pain"),1,0)

In [40]:
smileon_drop.head(30)

index  Unnamed: 0  patient_id  callreportnum reportversion  \
13   57743       57743     2384106       55264053      clinical   
19   57842       57842     2384744       55107015      clinical   
34   57667       57667     2384833       55320630      clinical   
39   57744       57744     2384922       55263902      clinical   
59   57335       57335     2384932       55699959      clinical   
79   57805       57805     2384949       55264227      clinical   
89   57723       57723     2384963       55270931      clinical   
102  57754       57754     2384971       55262425      clinical   
128  57760       57760     2384974       55260734      clinical   
126  57748       57748     2384974       55263079      clinical   
129  57668       57668     2384974       55320464      clinical   
140  54793       54793     2384995       57886463      clinical   
157  57812       57812     2385015       55117283      clinical   
170  57107       57107     2385046       55906577      clinical   
208  53625       53625     2385160       58732652      clinical   
219  22095       22095     2385160       77949729      clinical   
227  57618       57618     2385168       55352742      clinical   
240  57763       57763     2385321       55257550      clinical   
261  56889       56889     2385447       56208240      clinical   
262  20500       20500     2385447       79050156      clinical   
277  57765       57765     2385621       55256773      clinical   
270  10870       10870     2385621       86386343      clinical   
287  57739       57739     2386054       55266034      clinical   
283  54012       54012     2386054       58465132      clinical   
295  57487       57487     2386075       55504402      clinical   
309  57796       57796     2386386       55161958      clinical   
324  57699       57699     2386742       55295800      clinical   
333  56482       56482     2387558       56597836      clinical   
345  57669       57669     2387627       55320398      clinical   
355  46505       46505     2387646       63514080      clinical   

            callstart           callend  calllength  callstaff_id  \
13     8/7/2018 10:53    8/7/2018 10:55           2        116749   
19     8/1/2018 10:00    8/1/2018 10:22          22        118820   
34      8/9/2018 8:49     8/9/2018 8:50           1        118820   
39     8/7/2018 10:49    8/7/2018 10:50           1        116749   
59    8/23/2018 14:37   8/23/2018 15:05          28        118833   
79     8/2/2018 10:58    8/2/2018 11:16          18        116749   
89     8/7/2018 14:00    8/7/2018 14:04           4        118833   
102    8/7/2018 10:07    8/7/2018 10:08           1        116749   
128     8/7/2018 9:43     8/7/2018 9:44           1        118820   
126    8/7/2018 10:26    8/7/2018 10:27           1        118820   
129     8/9/2018 8:43     8/9/2018 8:45           2        118820   
140  11/12/2018 11:38  11/12/2018 11:41           3        118833   
157    8/1/2018 14:48    8/1/2018 14:48           0        118820   
170   8/31/2018 12:54   8/31/2018 12:56           2        118820   
208   12/14/2018 7:39   12/14/2018 7:41           2        118820   
219    8/26/2020 7:41    8/26/2020 7:42           1        118820   
227   8/10/2018 10:27   8/10/2018 10:37          10        118833   
240     8/7/2018 9:28     8/7/2018 9:34           6        116749   
261    9/12/2018 9:53    9/12/2018 9:57           4        118820   
262   9/23/2020 15:45   9/23/2020 15:47           2        140316   
277     8/7/2018 9:01     8/7/2018 9:07           6        116749   
270     4/1/2021 8:23     4/1/2021 8:25           2        118833   
287    8/7/2018 11:50    8/7/2018 11:51           1        116749   
283   12/4/2018 14:01   12/4/2018 14:04           3        118833   
295    8/16/2018 9:11    8/16/2018 9:13           2        116749   
309     8/3/2018 7:46     8/3/2018 8:33          47        118820   
324    8/8/2018 11:39    8/8/2018 11:46           7        11883

In [41]:
smileon.head()

index  Unnamed: 0  patient_id  callreportnum reportversion  \
11  57818       57818     2384106       55083365  registration   
13  57743       57743     2384106       55264053      clinical   
12  57851       57851     2384106       55628771      clinical   
3   56013       56013     2384106       57055663      clinical   
5   55239       55239     2384106       57578517      clinical   

           callstart           callend  calllength  callstaff_id  \
11    8/1/2018 13:22    8/1/2018 14:09          47        118833   
13    8/7/2018 10:53    8/7/2018 10:55           2        116749   
12     8/1/2018 8:53     8/1/2018 8:54           1        118821   
3    10/9/2018 18:59   10/9/2018 18:59           0        118821   
5   10/30/2018 17:09  10/30/2018 17:10           1        118821   

    entrystaff_id         entrydate Reviewed ReferralsMade patient_bday  \
11         118833   7/31/2018 14:09       No           NaN         1944   
13         116749    8/7/2018 10:55       No           NaN         1944   
12         118821    8/21/2018 8:54       No           NaN         1944   
3          118821  10/11/2018 18:59       No           NaN         1944   
5          118821  10/31/2018 17:10       No           NaN         1944   

            city      county state      zip demographic_household  \
11  Murfreesboro  Rutherford    TN  37130.0                     5   
13  Murfreesboro  Rutherford    TN  37130.0                     5   
12  Murfreesboro  Rutherford    TN  37130.0                     5   
3   Murfreesboro  Rutherford    TN  37130.0                     5   
5   Murfreesboro  Rutherford    TN  37130.0                     5   

   demographic_lang demographic_preferredlang demographic_ethnic  \
11              NaN                    Arabic                NaN   
13              NaN                    Arabic            Unknown   
12              NaN                    Arabic            Unknown   
3               NaN                    Arabic            Unknown   
5               NaN                    Arabic            Unknown   

   demographic_gender demographic_income demographic_howyouhearprogram  \
11             Female       $0 - $21 280           Dental Clinic Staff   
13             Female       $0 - $21 280           Dental Clinic Staff   
12             Female       $0 - $21 280           Dental Clinic Staff   
3              Female       $0 - $21 280           Dental Clinic Staff   
5              Female       $0 - $21 280           Dental Clinic Staff   

   demographic_otherassistance demographic_lastdentistvisit pe_insurance  \
11              SNAP; TennCare                 3 to 5 years           No   
13              SNAP; TennCare                 3 to 5 years           No   
12              SNAP; TennCare                 3 to 5 years           No   
3               SNAP; TennCare                 3 to 5 years           No   
5               SNAP; TennCare                 3 to 5 years           No   

   pe_smileonprequalify                pe_nextstep               pe_nextstep2  \
11                  Yes  Continued with Navigation                        NaN   
13                  Yes                        NaN  Continued with Navigation   
12                  Yes                        NaN  Continued with Navigation   
3                   Yes                        NaN  Continued with Navigation   
5                   Yes                        NaN  Continued with Navigation   

   ve_transitarranged ve_eligibility ve_verification_date  \
11                NaN            NaN                  NaN   
13                NaN            Yes             8/1/2018   
12                NaN            NaN             8/1/2018   
3                 NaN            NaN             8/1/2018   
5                 NaN            NaN             8/1/2018   

                              si_clinicname si_typewhereclientenrolled  \
11  Interfaith Dental Clinic - Murfreesboro                        NaN   
13                 

In [42]:
#excellent!  now - let's try and merge drop back with the main df so i can compare treatments with complaints
#a left join/merge should do it
smileon_test2 = smileon.merge(smileon_drop, how = 'left', on = 'callreportnum')

In [44]:
smileon_test2.head()

index_x  Unnamed: 0_x  patient_id_x  callreportnum reportversion_x  \
0    57818         57818       2384106       55083365    registration   
1    57743         57743       2384106       55264053        clinical   
2    57851         57851       2384106       55628771        clinical   
3    56013         56013       2384106       57055663        clinical   
4    55239         55239       2384106       57578517        clinical   

        callstart_x         callend_x  calllength_x  callstaff_id_x  \
0    8/1/2018 13:22    8/1/2018 14:09            47          118833   
1    8/7/2018 10:53    8/7/2018 10:55             2          116749   
2     8/1/2018 8:53     8/1/2018 8:54             1          118821   
3   10/9/2018 18:59   10/9/2018 18:59             0          118821   
4  10/30/2018 17:09  10/30/2018 17:10             1          118821   

   entrystaff_id_x       entrydate_x Reviewed_x ReferralsMade_x  \
0           118833   7/31/2018 14:09         No             NaN   
1           116749    8/7/2018 10:55         No             NaN   
2           118821    8/21/2018 8:54         No             NaN   
3           118821  10/11/2018 18:59         No             NaN   
4           118821  10/31/2018 17:10         No             NaN   

  patient_bday_x        city_x    county_x state_x    zip_x  \
0           1944  Murfreesboro  Rutherford      TN  37130.0   
1           1944  Murfreesboro  Rutherford      TN  37130.0   
2           1944  Murfreesboro  Rutherford      TN  37130.0   
3           1944  Murfreesboro  Rutherford      TN  37130.0   
4           1944  Murfreesboro  Rutherford      TN  37130.0   

  demographic_household_x demographic_lang_x demographic_preferredlang_x  \
0                       5                NaN                      Arabic   
1                       5                NaN                      Arabic   
2                       5                NaN                      Arabic   
3                       5                NaN                      Arabic   
4                       5                NaN                      Arabic   

  demographic_ethnic_x demographic_gender_x demographic_income_x  \
0                  NaN               Female         $0 - $21 280   
1              Unknown               Female         $0 - $21 280   
2              Unknown               Female         $0 - $21 280   
3              Unknown               Female         $0 - $21 280   
4              Unknown               Female         $0 - $21 280   

  demographic_howyouhearprogram_x demographic_otherassistance_x  \
0             Dental Clinic Staff                SNAP; TennCare   
1             Dental Clinic Staff                SNAP; TennCare   
2             Dental Clinic Staff                SNAP; TennCare   
3             Dental Clinic Staff                SNAP; TennCare   
4             Dental Clinic Staff                SNAP; TennCare   

  demographic_lastdentistvisit_x pe_insurance_x pe_smileonprequalify_x  \
0                   3 to 5 years             No                    Yes   
1                   3 to 5 years             No                    Yes   
2                   3 to 5 years             No                    Yes   
3                   3 to 5 years             No                    Yes   
4                   3 to 5 years             No                    Yes   

               pe_nextstep_x             pe_nextstep2_x ve_transitarranged_x  \
0  Continued with Navigation                        NaN                  NaN   
1                        NaN  Continued with Navigation                  NaN   
2                        NaN  Continued with Navigation                  NaN   
3                        NaN  Continued with Navigation                  NaN   
4                        NaN  Continued with Navigation                  NaN   

  ve_eligibility_x ve_verification_date_x  \
0              NaN                    NaN   
1              Yes               8/1/2018   
2              NaN       

In [47]:
#ah!  no!  duplicates everywhere.  i have a list of all my columns.  annoying, but this should work:
smileon = smileon.merge(smileon_drop, how = 'left', on = [
    'patient_id', 'callreportnum', 'reportversion', 'callstart', 'callend', 'calllength', 'callstaff_id', 'entrystaff_id', 'entrydate', 'Reviewed', 'ReferralsMade', 'patient_bday', 'city', 'county', 'state', 'zip', 'demographic_household', 'demographic_lang', 'demographic_preferredlang', 'demographic_ethnic', 'demographic_gender', 'demographic_income', 'demographic_howyouhearprogram', 'demographic_otherassistance', 'demographic_lastdentistvisit', 'pe_insurance', 'pe_smileonprequalify', 'pe_nextstep', 'pe_nextstep2', 've_transitarranged', 've_eligibility', 've_verification_date', 'si_clinicname', 'si_typewhereclientenrolled', 'blq_troubleeating', 'blq_dentalclinic', 'blq_patientreferralcommitment', 'blq_did_needhelp', 'blq_do_needhelp', 'blq_qualityoflife', 'blq_mouthcondition', 'blq_nextstep', 'blq_nextstep2', 's_num_lowerteeth', 's_num_upperteeth', 's_dentures', 's_dentureswhileeating', 's_problems12months', 's_brushing', 's_needforcare', 's_rootfrags', 's_lesions', 's_urgency', 's_decay', 's_lastscreening_date', 'ci_lastseen_date', 'ci_apptmade', 'ci_transitprovided', 'cmi_clinic', 'cmi_transitproblem', 'cmi_findingcareproblem', 'cmi_clientoutcome', 'oce_230', 'oce_1351', 'oce_1354', 'oce_2140', 'oce_2150', 'oce_2160', 'oce_2161', 'oce_2330', 'oce_2331', 'oce_2332', 'oce_2335', 'oce_2390', 'oce_2391', 'oce_2392', 'oce_2393', 'oce_2394', 'oce_2915', 'oce_2920', 'oce_2933', 'oce_2940', 'oce_2950', 'oce_2954', 'oce_2980', 'oce_3220', 'oce_4341', 'oce_4342', 'oce_6930', 'oce_6980', 'oce_7111', 'oce_7140', 'oce_7210', 'oce_7220', 'oce_7230', 'oce_7240', 'oce_7241', 'oce_7250', 'oce_7261', 'oce_7286', 'oce_7310', 'oce_7311', 'oce_7320', 'oce_7321', 'oce_7410', 'oce_7411', 'oce_7471', 'oce_7510', 'oce_7971', 'oce_7972', 'oce_9110', 'oce_9951', 'oce_WHATCAREGIVEN', 'tp_plancomplete', 'tp_plandeveloped', 'tp_clientrestored'
])

In [48]:
smileon.head()

index_x  Unnamed: 0_x  patient_id  callreportnum reportversion  \
0    57818         57818     2384106       55083365  registration   
1    57743         57743     2384106       55264053      clinical   
2    57851         57851     2384106       55628771      clinical   
3    56013         56013     2384106       57055663      clinical   
4    55239         55239     2384106       57578517      clinical   

          callstart           callend  calllength  callstaff_id  \
0    8/1/2018 13:22    8/1/2018 14:09          47        118833   
1    8/7/2018 10:53    8/7/2018 10:55           2        116749   
2     8/1/2018 8:53     8/1/2018 8:54           1        118821   
3   10/9/2018 18:59   10/9/2018 18:59           0        118821   
4  10/30/2018 17:09  10/30/2018 17:10           1        118821   

   entrystaff_id         entrydate Reviewed ReferralsMade patient_bday  \
0         118833   7/31/2018 14:09       No           NaN         1944   
1         116749    8/7/2018 10:55       No           NaN         1944   
2         118821    8/21/2018 8:54       No           NaN         1944   
3         118821  10/11/2018 18:59       No           NaN         1944   
4         118821  10/31/2018 17:10       No           NaN         1944   

           city      county state      zip demographic_household  \
0  Murfreesboro  Rutherford    TN  37130.0                     5   
1  Murfreesboro  Rutherford    TN  37130.0                     5   
2  Murfreesboro  Rutherford    TN  37130.0                     5   
3  Murfreesboro  Rutherford    TN  37130.0                     5   
4  Murfreesboro  Rutherford    TN  37130.0                     5   

  demographic_lang demographic_preferredlang demographic_ethnic  \
0              NaN                    Arabic                NaN   
1              NaN                    Arabic            Unknown   
2              NaN                    Arabic            Unknown   
3              NaN                    Arabic            Unknown   
4              NaN                    Arabic            Unknown   

  demographic_gender demographic_income demographic_howyouhearprogram  \
0             Female       $0 - $21 280           Dental Clinic Staff   
1             Female       $0 - $21 280           Dental Clinic Staff   
2             Female       $0 - $21 280           Dental Clinic Staff   
3             Female       $0 - $21 280           Dental Clinic Staff   
4             Female       $0 - $21 280           Dental Clinic Staff   

  demographic_otherassistance demographic_lastdentistvisit pe_insurance  \
0              SNAP; TennCare                 3 to 5 years           No   
1              SNAP; TennCare                 3 to 5 years           No   
2              SNAP; TennCare                 3 to 5 years           No   
3              SNAP; TennCare                 3 to 5 years           No   
4              SNAP; TennCare                 3 to 5 years           No   

  pe_smileonprequalify                pe_nextstep               pe_nextstep2  \
0                  Yes  Continued with Navigation                        NaN   
1                  Yes                        NaN  Continued with Navigation   
2                  Yes                        NaN  Continued with Navigation   
3                  Yes                        NaN  Continued with Navigation   
4                  Yes                        NaN  Continued with Navigation   

  ve_transitarranged ve_eligibility ve_verification_date  \
0                NaN            NaN                  NaN   
1                NaN            Yes             8/1/2018   
2                NaN            NaN             8/1/2018   
3                NaN            NaN             8/1/2018   
4                NaN            NaN             8/1/2018   

                             si_clinicname si_typewhereclientenrolled  \
0  Interfaith Dental Clinic - Murfreesboro                        NaN   
1                                      NaN        

In [58]:
smileon = smileon.drop(['Unnamed: 0_x'], axis =1)

In [59]:
#resorting back into order:
smileon = smileon.sort_values(['patient_id', 'callreportnum'])

In [60]:
smileon.head()

index_x  patient_id  callreportnum reportversion         callstart  \
0    57818     2384106       55083365  registration    8/1/2018 13:22   
1    57743     2384106       55264053      clinical    8/7/2018 10:53   
2    57851     2384106       55628771      clinical     8/1/2018 8:53   
3    56013     2384106       57055663      clinical   10/9/2018 18:59   
4    55239     2384106       57578517      clinical  10/30/2018 17:09   

            callend  calllength  callstaff_id  entrystaff_id  \
0    8/1/2018 14:09          47        118833         118833   
1    8/7/2018 10:55           2        116749         116749   
2     8/1/2018 8:54           1        118821         118821   
3   10/9/2018 18:59           0        118821         118821   
4  10/30/2018 17:10           1        118821         118821   

          entrydate Reviewed ReferralsMade patient_bday          city  \
0   7/31/2018 14:09       No           NaN         1944  Murfreesboro   
1    8/7/2018 10:55       No           NaN         1944  Murfreesboro   
2    8/21/2018 8:54       No           NaN         1944  Murfreesboro   
3  10/11/2018 18:59       No           NaN         1944  Murfreesboro   
4  10/31/2018 17:10       No           NaN         1944  Murfreesboro   

       county state      zip demographic_household demographic_lang  \
0  Rutherford    TN  37130.0                     5              NaN   
1  Rutherford    TN  37130.0                     5              NaN   
2  Rutherford    TN  37130.0                     5              NaN   
3  Rutherford    TN  37130.0                     5              NaN   
4  Rutherford    TN  37130.0                     5              NaN   

  demographic_preferredlang demographic_ethnic demographic_gender  \
0                    Arabic                NaN             Female   
1                    Arabic            Unknown             Female   
2                    Arabic            Unknown             Female   
3                    Arabic            Unknown             Female   
4                    Arabic            Unknown             Female   

  demographic_income demographic_howyouhearprogram  \
0       $0 - $21 280           Dental Clinic Staff   
1       $0 - $21 280           Dental Clinic Staff   
2       $0 - $21 280           Dental Clinic Staff   
3       $0 - $21 280           Dental Clinic Staff   
4       $0 - $21 280           Dental Clinic Staff   

  demographic_otherassistance demographic_lastdentistvisit pe_insurance  \
0              SNAP; TennCare                 3 to 5 years           No   
1              SNAP; TennCare                 3 to 5 years           No   
2              SNAP; TennCare                 3 to 5 years           No   
3              SNAP; TennCare                 3 to 5 years           No   
4              SNAP; TennCare                 3 to 5 years           No   

  pe_smileonprequalify                pe_nextstep               pe_nextstep2  \
0                  Yes  Continued with Navigation                        NaN   
1                  Yes                        NaN  Continued with Navigation   
2                  Yes                        NaN  Continued with Navigation   
3                  Yes                        NaN  Continued with Navigation   
4                  Yes                        NaN  Continued with Navigation   

  ve_transitarranged ve_eligibility ve_verification_date  \
0                NaN            NaN                  NaN   
1                NaN            Yes             8/1/2018   
2                NaN            NaN             8/1/2018   
3                NaN            NaN             8/1/2018   
4                NaN            NaN             8/1/2018   

                             si_clinicname si_typewhereclientenrolled  \
0  Interfaith Dental Clinic - Murfreesboro                        NaN   
1                                      NaN                        NaN   
2  Interfaith Dental Clinic - Murfreesboro                   

In [ ]:
#let's play with this dtale tool that robert found!  
#d=dtale.show(smileon)
#d.open_browser()